barretenberg/cpp/src/barretenberg/polynomials/pow.hpp

```diff
@@ -1,32 +1,21 @@
 #pragma once
 
+#include <cstddef>
+#include <vector>
 namespace barretenberg {
 
 /**
  * @brief Succinct representation of the `pow` polynomial that can be partially evaluated variable-by-variable.
- * pow(X) = ∏_{0≤l<d} ((1−X_l) + X_l⋅ζ_l)
+ * pow(X_0,X_1,..,X_d) = \prod_{0≤l<d} ((1−X_l) + X_l⋅β_l) for a vector {β_0,...,β_{d-1}}
  *
  * @details Let
  * - d be the number of variables
  * - l be the current Sumcheck round ( l ∈ {0, …, d-1} )
- * - u_{0}, ..., u_{l-1} the challenges sent by the verifier in rounds 0 to l-1.
+ * - u_0, ..., u_l-1 the challenges sent by the verifier in rounds 0 to l-1.
  *
- * We define
- *
- * - ζ_{0}, ..., ζ_{d-1}, as ζ_{l} = ζ^{ 2^l }.
- *   When 0 ≤ i < 2^d is represented in bits [i_{0}, ..., i_{d-1}] where i_{0} is the MSB, we have
- *   ζ^{i} = ζ^{ ∑_{0≤l<d} i_l ⋅ 2^l }
- *         =     ∏_{0≤l<d} ζ^{ i_l ⋅ 2^l }
- *         =     ∏_{0≤l<d} ζ_l^{ i_l }
- *         =     ∏_{0≤l<d} { ( 1-i_l ) + i_l ⋅ ζ_l } // As i_{l} \in \{0, 1\}
- *   Note that
- *   - ζ_{0} = ζ,
- *   - ζ_{l+1} = ζ_{l}^2,
- *   - ζ_{d-1}   = ζ^{ 2^{d-1} }
- *
- * - pow(X) = ∏_{0≤l<d} ((1−X_l) + X_l⋅ζ_l) is the multi-linear polynomial whose evaluation at the i-th index
- *   of the full hypercube, equals ζⁱ.
- *   We can also see it as the multi-linear extension of the vector (1, ζ, ζ², ...).
+ * - pow(X) = ∏_{0≤l<d} ((1−X_l) + X_l⋅β_l) is the multilinear polynomial whose evaluation at the i-th index
+ *   of the full hypercube, equals ∏ β_j where j are the bits set to 1 in the binary representation of i
+ *   We can also see it as the multi-linear extension of the vector \vec{β}.
  *
  * - At round l, we iterate over all remaining vertices (i_{l+1}, ..., i_{d-1}) ∈ {0,1}^{d-l-1}.
```

Comment:
Round 0: $2^d$ vertices.
Round 1: $(2^d)/2 = 2^{d-1}$ vertices
...
Round $\ell$: $2^{d-\ell}$ vertices, right? Are you off by one?

I find this a little confusing. You're taking about sumcheck right? Rather than "remaining vertices" I'd just say we iterate over a boolean hypercube of $\dim$ $(d-l)$ (this is naturally a subcube of the full $\dim$ $d$ one). Based on how the execution trace fold, it's more like 'the start' of the full cube. -- or convince me otherwise!

Reply:
Remaining vertices was a term already in the description, but your suggestion is definitely better, thanks, Also the off by one was in master but yeah it is an off by one

Reply:
Tried to further clarify things.